superdesk/publish/formatters/__init__.py

```diff
@@ -36,6 +37,24 @@ def can_format(self, format_type, article):
         """Test if formatter can format for given article."""
         raise NotImplementedError()
 
+    def append_public_service_announcements(self, article):
+        """
+        Checks if the article has any Public Service Announcements and if available appends each of them to the body.
+
+        :return: body with public service announcements.
+        """
+
+        body = ''
+        if article[ITEM_TYPE] in [CONTENT_TYPE.TEXT, CONTENT_TYPE.PREFORMATTED]:
+            body = article.get('body_html', '')
+        elif article[ITEM_TYPE] in [CONTENT_TYPE.AUDIO, CONTENT_TYPE.PICTURE, CONTENT_TYPE.VIDEO]:
+            body = article.get('description', '')
+
+        if body and article.get('psa_footer'):
+            body = '{}<br>{}'.format(body, article.get('psa_footer'))
```

Comment:
maybe we can call this field `body_footer` ? it would match the ninjs spec and would be more generic for core